packages/flutter/lib/src/painting/_network_image_web.dart

```diff
@@ -121,17 +222,17 @@ class NetworkImage
     // We use a different method when headers are set because the
     // `ui.webOnlyInstantiateImageCodecFromUrl` method is not capable of handling headers.
     if (key.headers?.isNotEmpty ?? false) {
-      final Completer<html.HttpRequest> completer =
-          Completer<html.HttpRequest>();
-      final html.HttpRequest request = httpRequestFactory();
+      final Completer<DomXMLHttpRequest> completer =
+          Completer<DomXMLHttpRequest>();
+      final DomXMLHttpRequest request = httpRequestFactory();
 
-      request.open('GET', key.url, async: true);
+      request.open('GET', key.url, true);
       request.responseType = 'arraybuffer';
       key.headers!.forEach((String header, String value) {
         request.setRequestHeader(header, value);
       });
 
-      request.onLoad.listen((html.ProgressEvent e) {
+      request.addEventListener('load', allowInterop((DomEvent e) {
```

Comment:
It is a `DomProgressEvent`, but we have to use the supertype here because `EventTarget.addEventListener` takes an `EventListener` function` i.e. `void Function(DomEvent)`.

Given these types are erased anyways, we could probably find a way to change the type without breaking anything, but I'm not sure its worth it.

I'm going to assume we want this to continue to reflect the MDN, but please let me know if you'd prefer we investigate tighter type bounds.